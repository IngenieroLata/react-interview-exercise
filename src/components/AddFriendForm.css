:local(.form) {
  background-color: white;
  border-bottom: 1px solid #abaaaa;
  padding-bottom: 1em;
}

:local(.addFriendInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
}

.btn-primary {
  color: #fff;
  background-color: #5c75b0;
  border-color: #5c75b0;
}
