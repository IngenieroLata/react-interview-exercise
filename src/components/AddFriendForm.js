import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './AddFriendForm.css';

class AddFriendForm extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      name: '',
      gender: ''
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  render() {
    const { name, gender } = this.state;
    return (
      <form action="" onSubmit={this.handleSubmit} className={styles.form}>
        <div className="form-group">
          <input
            type="text"
            autoFocus="true"
            name="name"
            className={classnames('form-control', styles.addFriendInput)}
            placeholder="Type the name of a friend"
            value={name}
            onChange={this.handleChange}
          />
        </div>

        <label className="radio-inline">
          <input
            type="radio"
            name="gender"
            value="male"
            checked={gender === 'male'}
            onChange={this.handleChange}
          />
          Male
        </label>
        <label className="radio-inline">
          <input
            type="radio"
            name="gender"
            value="female"
            checked={gender === 'female'}
            onChange={this.handleChange}
          />
          Female
        </label>
        <button type="submit" className="pull-right btn btn-primary btn-xs">
          Add Friend
        </button>

        <div className="clearfix" />
      </form>
    );
  }

  handleChange({ target: { name, value } }) {
    this.setState({ [name]: value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.addFriend(this.state);
    this.setState({ name: '', gender: '' });
  }
}

AddFriendForm.propTypes = {
  addFriend: PropTypes.func.isRequired
};

export default AddFriendForm;
