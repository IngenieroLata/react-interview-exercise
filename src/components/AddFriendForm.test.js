import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AddFriendForm from './AddFriendForm';

const props = {
  addFriend: jest.fn()
};

describe('<AddFriendForm />', () => {
  it('should control radio buttons', () => {
    const wrapper = shallow(<AddFriendForm {...props} />);
    wrapper
      .find('[value="male"]')
      .simulate('change', { target: { value: 'male', name: 'gender' } });
    expect(wrapper.state('gender')).toBe('male');
  });

  it('should control input text', () => {
    const wrapper = shallow(<AddFriendForm {...props} />);
    wrapper
      .find('[name="name"]')
      .simulate('change', { target: { value: 'Jon Doe', name: 'name' } });
    expect(wrapper.state('name')).toBe('Jon Doe');
  });

  it('should reset form values on submit', () => {
    const wrapper = shallow(<AddFriendForm {...props} />);
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(wrapper.state()).toEqual({
      name: '',
      gender: ''
    });
  });

  it('should call creation action', () => {
    const wrapper = shallow(<AddFriendForm {...props} />);
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(props.addFriend).toHaveBeenCalled();
  });

  it('Matches the snapshot render', () => {
    const wrapper = shallow(<AddFriendForm {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
