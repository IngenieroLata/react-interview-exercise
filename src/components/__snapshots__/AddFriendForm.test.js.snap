exports[`<AddFriendForm /> Matches the snapshot render 1`] = `
<form
  action=""
  onSubmit={[Function]}>
  <div
    className="form-group">
    <input
      autoFocus="true"
      className="form-control"
      name="name"
      onChange={[Function]}
      placeholder="Type the name of a friend"
      type="text"
      value="" />
  </div>
  <label
    className="radio-inline">
    <input
      checked={false}
      name="gender"
      onChange={[Function]}
      type="radio"
      value="male" />
    Male
  </label>
  <label
    className="radio-inline">
    <input
      checked={false}
      name="gender"
      onChange={[Function]}
      type="radio"
      value="female" />
    Female
  </label>
  <button
    className="pull-right btn btn-primary btn-xs"
    type="submit">
    Extra small button
  </button>
  <div
    className="clearfix" />
</form>
`;
